pkg/dcgmexporter/const.go

```diff
@@ -25,26 +25,28 @@ const (
 	DCGMXIDErrorsCount DCGMExporterMetric = iota + 9000
 )
 
+// DCGMFields maps DCGMExporterMetric String to enum
+var DCGMFields = map[string]DCGMExporterMetric{
+	DCGMXIDErrorsCount.String(): DCGMXIDErrorsCount,
+	DCGMFIUnknown.String():      DCGMFIUnknown,
+}
+
 // String method to convert the enum value to a string
-func (enm DCGMExporterMetric) String() string {
-	switch enm {
+func (d DCGMExporterMetric) String() string {
+	switch d {
 	case DCGMXIDErrorsCount:
 		return "DCGM_EXP_XID_ERRORS_COUNT"
 	default:
 		return "DCGM_FI_UNKNOWN"
 	}
 }
 
-func mustParseDCGMExporterMetric(s string) DCGMExporterMetric {
-	metrics := map[string]DCGMExporterMetric{
-		DCGMXIDErrorsCount.String(): DCGMXIDErrorsCount,
-		DCGMFIUnknown.String():      DCGMFIUnknown,
-	}
-	mv, ok := metrics[s]
+func IdentifyMetricType(s string) (DCGMExporterMetric, error) {
+	mv, ok := DCGMFields[s]
 	if !ok {
-		panic(fmt.Sprintf(`cannot parse:[%s] as DCGMExporterMetric`, s))
+		return mv, fmt.Errorf("unknown DCGMExporterMetric field '%s'", s)
```

Comment:
The produced error message will not be in a consistency with another error message.

Reply:
Reverted to original format in favor of a new PR to make error messages consistent.